modules/apps/asset/asset-info-impl/src/main/java/com/liferay/asset/info/internal/item/provider/AssetEntryInfoItemFieldSetProviderImpl.java

```diff
@@ -115,41 +194,63 @@ public InfoFieldSet getInfoFieldSet(String itemClassName) {
 		}
 	}
 
-	private String _getCategoryNames(AssetEntry assetEntry) {
-		Locale locale = LocaleThreadLocal.getThemeDisplayLocale();
+	private Set<AssetVocabulary> _getAssetVocabularies(AssetEntry assetEntry) {
+		List<AssetCategory> assetCategories = assetEntry.getCategories();
+		Set<AssetVocabulary> assetVocabularies = new HashSet<>();
 
-		return ListUtil.toString(
-			assetEntry.getCategories(),
-			new Accessor<AssetCategory, String>() {
+		for (AssetCategory assetCategory : assetCategories) {
+			AssetVocabulary assetVocabulary =
+				_assetVocabularyLocalService.fetchAssetVocabulary(
+					assetCategory.getVocabularyId());
 
-				@Override
-				public String get(AssetCategory assetCategory) {
-					String title = assetCategory.getTitle(locale);
+			assetVocabularies.add(assetVocabulary);
+		}
 
-					if (Validator.isNull(title)) {
-						return assetCategory.getName();
-					}
+		return assetVocabularies;
+	}
 
-					return title;
-				}
+	private String _getCategoryNames(AssetEntry assetEntry, long vocabularyId) {
```

Comment:
This method is doing three things: (1) obtain the categories (2) filter them by vocabulary and (3) convert the list to a map.
I think it would be best to split 2 and 3 into separate methods (since 1 is just a simple call) so that instead of invoking it with vocabularyId=0 when filtering is not desired, the filter method is just not invoked. The invoking code would look something like this:
````
getCategoryNames(filterByVocabulary(assetEntry.getCategories(), vocabularyId)
````

or

````
getCategoryNames(assetEntry.getCategories())
````

That way it's much more clear to anyone reading the code when filtering is happening and when not. Right now it requires reading the implementation of _getCategoryNames() to find out what passing 0 means.